clio-client/src/main/scala/org/broadinstitute/clio/client/webclient/ClioWebClient.scala

```diff
@@ -98,57 +101,95 @@ class ClioWebClient(
+         *
+         * DON'T refresh if the underlying access token has no expiration time,
+         * because refresh might not be supported for that token.
+         */
+        val maybeToken = Option(creds.getAccessToken)
+        lazy val maybeTokenExpirationInstant = for {
+          token <- maybeToken
+          expiration <- Option(token.getExpirationTime)
+        } yield {
+          expiration.toInstant
+        }
+        val shouldRefresh =
+          maybeToken.isEmpty || maybeTokenExpirationInstant.exists(
+            _.isAfter(now)
+          )
+
+        if (shouldRefresh) {
+          creds.refresh()
         }
-        case QueueOfferResult.Failure(ex) =>
-          responsePromise.failure(ex)
-        case QueueOfferResult.QueueClosed =>
-          responsePromise.failure {
-            new RuntimeException(
-              s"Client queue was closed while adding the request: $request"
-            )
-          }
       }
 
-      responsePromise.future
-        .logErrorMsg("Failed to send HTTP request to Clio server")
-        .flatMap(ensureOkResponse)
+      request.addCredentials(
+        OAuth2BearerToken(creds.getAccessToken.getTokenValue)
+      )
+    }
+
+    requestStream.offer(requestWithCreds -> responsePromise).flatMap {
+      queueResult =>
+        queueResult match {
+          case QueueOfferResult.Enqueued => ()
+          case QueueOfferResult.Dropped => {
+            responsePromise.failure {
+              new RuntimeException(
+                s"Client queue was too full to add the request: $request"
+              )
+            }
+          }
+          case QueueOfferResult.Failure(ex) =>
+            responsePromise.failure(ex)
+          case QueueOfferResult.QueueClosed =>
+            responsePromise.failure {
+              new RuntimeException(
+                s"Client queue was closed while adding the request: $request"
+              )
+            }
+        }
+
+        responsePromise.future
+          .logErrorMsg("Failed to send HTTP request to Clio server")
+          .flatMap(ensureOkResponse)
     }
   }
 
   def getClioServerVersion: Future[Json] = {
-    dispatchRequest(HttpRequest(uri = "/version"))
+    dispatchRequest(HttpRequest(uri = "/version"), None)
       .flatMap(unmarshal[Json])
   }
 
   def getClioServerHealth: Future[Json] = {
-    dispatchRequest(HttpRequest(uri = "/health"))
+    dispatchRequest(HttpRequest(uri = "/health"), None)
       .flatMap(unmarshal[Json])
   }
 
   def getSchema(
     transferIndex: TransferIndex
-  )(implicit credentials: HttpCredentials): Future[Json] = {
+  )(implicit credentials: OAuth2Credentials): Future[Json] = {
     dispatchRequest(
-      HttpRequest(uri = s"/api/v1/${transferIndex.urlSegment}/schema")
-        .addHeader(Authorization(credentials = credentials))
+      HttpRequest(uri = s"/api/v1/${transferIndex.urlSegment}/schema"),
+      Some(credentials)
```

Comment:
I thought the convention was to use `Option()` when constructing from a variable to avoid `Some(null)`